qiskit/qasm3/exporter.py

```diff
@@ -1145,9 +1145,11 @@ def _rebind_scoped_parameters(self, expression):
         # missing, pending a new system in Terra to replace it (2022-03-07).
         if not isinstance(expression, ParameterExpression):
             return expression
+        if isinstance(expression, Parameter):
+            return self.symbols.get_variable(expression).string
```

Comment:
Is this a different behavior here? I've not looked too much at `ParameterExpression`, but is it odd for this to return a string rather than an `Expression`? Perhaps I'm missing something 🙂

Reply:
You're not missing anything, it's just abuse of the complete absence of handling for `ParameterExpression` in the exporter - we literally just call `str` and hope for the best.

The reason not to construct a `ParameterExpression` is because the case of a standalone `Parameter` is _very_ common on the performance critical path, and `ParameterExpression` (and `Parameter`) are both very expensive to construct.